--- List.pm.orig	2009-12-18 13:50:25.000000000 -0800
+++ src/lib/List.pm	2009-12-18 13:50:45.000000000 -0800
@@ -2623,6 +2623,9 @@
     foreach my $i (@{$self->{'admin'}{'custom_header'}}) {
 	$hdr->add($1, $2) if ($i=~/^([\S\-\:]*)\s(.*)$/);
     }
+
+    ## Add Sender: Header for DKIM compliance
+    $hdr->add('Sender', $name.&Conf::get_robot_conf($robot, 'return_path_suffix').'@'.$host);
     
     ## Add RFC 2919 header field
     if ($hdr->get('List-Id')) {
