--- orig/wwsympa/wwsympa.fcgi.in	2009-09-23 08:44:56.000000000 -0700
+++ wwsympa/wwsympa.fcgi.in	2009-11-16 07:30:24.000000000 -0800
@@ -10110,7 +10110,8 @@
 
      # set list status to pending if creation list is moderated
      if ($r_action =~ /listmaster/) {
-	 $list->{'admin'}{'status'} = 'pending' ;
+	 # do not set status to pending, per riseup sysadmin
+	 #$list->{'admin'}{'status'} = 'pending' ;
 	 &List::send_notify_to_listmaster('request_list_renaming',$robot, 
 					  {'list' => $list,
 					   'new_listname' => $in{'new_listname'},
