Index: src/List.pm
===================================================================
--- src/List.pm	(revision 51)
+++ src/lib/List.pm	(working copy)
@@ -2621,6 +2621,9 @@
 	$hdr->add($1, $2) if ($i=~/^([\S\-\:]*)\s(.*)$/);
     }
     
+    ## Add Sender: Header for DKIM compliance
+    $hdr->add('Sender', $name.&Conf::get_robot_conf($robot, 'return_path_suffix').'@'.$host);
+
     ## Add RFC 2919 header field
     if ($hdr->get('List-Id')) {
 	&do_log('notice', 'Found List-Id: %s', $hdr->get('List-Id'));
