Index: wwsympa/wwsympa.fcgi
===================================================================
--- wwsympa/wwsympa.fcgi	(revision 51)
+++ wwsympa/wwsympa.fcgi	(working copy)
@@ -10106,7 +10106,10 @@
 
      # set list status to pending if creation list is moderated
      if ($r_action =~ /listmaster/) {
-	 $list->{'admin'}{'status'} = 'pending' ;
+
+   	 # [XXX] As per request of the riseup sysadmins, DO NOT CHANGE to pending
+	 # $list->{'admin'}{'status'} = 'pending' ;
+
 	 &List::send_notify_to_listmaster('request_list_renaming',$robot, 
 					  {'list' => $list,
 					   'new_listname' => $in{'new_listname'},
