--- sympa-5.3.3/src/List.pm    2007-06-26 18:09:52.000000000 +0200
+++ src/List.pm     2007-11-18 21:10:59.000000000 +0100
@@ -4101,7 +4101,7 @@
     &do_log('debug2', 'List::send_notify_to_owner(%s, %s)', $self->{'name'}, $operation);

     my $host = $self->{'admin'}{'host'};
-    my @to = $self->get_owners_email;
+    my @to = $self->get_owners_email();
     my $robot = $self->{'domain'};

     unless (@to) {
