--- /tmp/wwsympa.fcgi		2007-12-08 08:26:23.000000000 -0800
+++ wwsympa/wwsympa.fcgi	2007-12-08 09:02:18.942143372 -0800
@@ -5100,6 +5100,17 @@
 	 return 'loginrequest';
      }
 
+     ## [XXX] If a list is not 'open' returns undef - there may be a better
+     #  thing to return
+     unless ($list->{'admin'}{'status'} eq 'open') {
+     &report::reject_report_web('user','list_not_open',{'status' =>  $list->{'admin'}{'status'}},$param->{'action'});
+     &wwslog('info','do_add: list not open');
+     &web_db_log({'target_email' => $in{'email'}||$in{'pending_email'},
+              'status' => 'error',
+              'error_type' => 'list_not_open'});
+         return undef;
+     }
+     
      my $email_regexp = &tools::get_regexp('email');
      if ($in{'dump'}) {
 	 foreach (split /\n/, $in{'dump'}) {
--- /tmp/web_tt2/error.tt2	2007-12-08 08:26:22.000000000 -0800
+++ web_tt2/error.tt2	2007-12-08 09:04:33.664731559 -0800
@@ -115,6 +115,7 @@
   [% ELSIF u_err.msg == 'select_month' %][%|loc%]Please select archive months[%END%]
   [% ELSIF u_err.msg == 'no_soap_service' %][%|loc%]No SOAP service[%END%]
   [% ELSIF u_err.msg == 'auth_failed' %][%|loc%]Authentication failed[%END%]
+  [% ELSIF u_err.msg == 'list_not_open' %][%|loc(u_err.status)%]List status is '%1'[%END%]
   [% ELSIF u_err.msg %][%u_err.msg%]
   [% END %]
 <br />		
