--- ./List.pm	2007-12-10 08:51:25.000000000 -0800
+++ src/List.pm	2007-12-22 08:57:36.901842876 -0800
@@ -2459,7 +2459,7 @@
     $hdr->add('X-Loop', "$name\@$host");
     $hdr->add('X-Sequence', $sequence);
     $hdr->add('Errors-to', $name.&Conf::get_robot_conf($robot, 'return_path_suffix').'@'.$host);
-    $hdr->add('Precedence', 'list');
+    $hdr->add('Precedence', 'bulk');
     $hdr->add('X-no-archive', 'yes');
     foreach my $i (@{$self->{'admin'}{'custom_header'}}) {
 	$hdr->add($1, $2) if ($i=~/^([\S\-\:]*)\s(.*)$/);
