--- /home/kclair/sympa-5.3.4/src/sympa.pl	2009-08-14 10:34:24.000000000 -0700
+++ src/sympa.pl	2009-08-17 09:31:05.000000000 -0700
@@ -162,7 +162,6 @@
 
 $log_level = $main::options{'log_level'} if ($main::options{'log_level'}); 
 
-my @parser_param = ($*, $/);
 my %loop_info;
 my %msgid_table;
 
@@ -910,9 +909,6 @@
 	my $list;
 	my ($t_listname, $t_robot);
 	
-	# trying to fix a bug (perl bug ??) of solaris version
-	($*, $/) = @parser_param;
-
 	## test ever if it is an old bad file
 	if ($t_filename =~ /^BAD\-/i){
  	    if ((stat "$t_spool/$t_filename")[9] < (time - &Conf::get_robot_conf($robot, 'clean_delay_queue')*86400) ){
--- /home/kclair/sympa-5.3.4/src/parser.pl	2004-04-08 01:51:40.000000000 -0700
+++ src/parser.pl	2009-08-17 09:33:35.000000000 -0700
@@ -60,8 +60,8 @@
     my ($old_index, $old_data) = ($index, $data);
     my @old_t = @t;
 
-    my @old_mode = ($*, $/);
-    ($*, $/) = (0, "\n");
+    my @old_mode = ($/);
+    ($/) = (0, "\n");
 
     my $old_desc;
     if (ref($output) eq 'ARRAY') {           
@@ -104,7 +104,7 @@
 	select $old_desc;
     }
     
-    ($*, $/) = @old_mode;
+    ($/) = @old_mode;
 
     ($index, $data) = ($old_index, $old_data);
     @t = @old_t;
