diff --git a/sympa-6.1.14-src/src/lib/List.pm b/sympa-6.1.14-src/src/lib/List.pm
index 4e95a59..4b855a4 100644
--- a/sympa-6.1.14-src/src/lib/List.pm
+++ b/sympa-6.1.14-src/src/lib/List.pm
@@ -11415,6 +11415,7 @@ sub compute_topic {
 	    my $charset = $part->head->mime_attr("Content-Type.Charset");
 	    $charset = MIME::Charset->new($charset);
 	    if (defined $part->bodyhandle) {
+              if (defined $msg->bodyhandle) {
 		my $body = $msg->bodyhandle->as_string();
 		my $converted;
 		eval {
@@ -11424,6 +11425,9 @@ sub compute_topic {
 		    $converted = Encode::decode('US-ASCII', $body);
 		}
 		$mail_string .= $converted."\n";
+              }else {
+                  &do_log('err','message %s does not have a bodyhandle?',$msg_id);
+              }
 	    }
 	}
     }
