--- pristine/sympa-5.3.3/src/List.pm    2007-06-26 18:09:52.000000000 +0200
+++ sympa-5.3.3/src/List.pm     2007-11-18 17:53:23.000000000 +0100
@@ -2458,6 +2458,9 @@
     foreach my $i (@{$self->{'admin'}{'custom_header'}}) {
        $hdr->add($1, $2) if ($i=~/^([\S\-\:]*)\s(.*)$/);
     }
+
+    ## Add Sender: Header for DKIM compliance
+    $hdr->add('Sender', $name.&Conf::get_robot_conf($robot, 'return_path_suffix').'@'.$host);

     ## Add RFC 2919 header field
     if ($hdr->get('List-Id')) {
