Index: src/List.pm
===================================================================
--- src/List.pm	(revision 4590)
+++ src/List.pm	(working copy)
@@ -2865,7 +2865,7 @@
     }
 
     my $old = $/;
-    $/ = "\n\n" . &tools::get_separator() . "\n\n";
+    local $/ = "\n\n" . &tools::get_separator() . "\n\n";
     
     ## Digest split in individual messages
     open DIGEST, $filename or return undef;
@@ -2891,7 +2891,7 @@
 	push @list_of_mail, $mail;
     }
     close DIGEST;
-    $/ = $old;
+    local $/ = $old;
 
     ## Deletes the introduction part
     splice @list_of_mail, 0, 1;
@@ -8185,7 +8185,7 @@
 	}
 	
 	## Just in case...
-	$/ = "\n";
+	local $/ = "\n";
 	
 	## Split in paragraphs
 	my $i = 0;
@@ -9933,7 +9933,7 @@
     my (@paragraphs);
 
     ## Just in case...
-    $/ = "\n";
+    local $/ = "\n";
 
     ## Set defaults to 1
     foreach my $pname (keys %::pinfo) {       
Index: src/Family.pm
===================================================================
--- src/Family.pm	(revision 4590)
+++ src/Family.pm	(working copy)
@@ -2746,7 +2746,7 @@
     my $error = 0;
 
     ## Just in case...
-    $/ = "\n";
+    local $/ = "\n";
 
     while (<FILE>) {
 	next if /^\s*(\#.*|\s*)$/;
Index: src/Conf.pm
===================================================================
--- src/Conf.pm	(revision 4590)
+++ src/Conf.pm	(working copy)
@@ -1227,7 +1227,7 @@
     my (@paragraphs);
     
     ## Just in case...
-    $/ = "\n";
+    local $/ = "\n";
     
     ## Set defaults to 1
     foreach my $pname (keys %structure) {       
Index: wwsympa/bounce-lib.pl
===================================================================
--- wwsympa/bounce-lib.pl	(revision 4590)
+++ wwsympa/bounce-lib.pl	(working copy)
@@ -27,7 +27,7 @@
 sub rfc1891 {
     my ($fic, $result, $from) = @_;
   
-    $/ = "\n";
+    local $/ = "\n";
 
     my $nbrcpt;
     my $parser = new MIME::Parser;
@@ -67,7 +67,7 @@
 	    exit;
 	}else {
 	    ## Multiline paragraph separator
-	    $/ = '';
+	    local $/ = '';
 
 	    while (<BODY>) {
 
@@ -91,7 +91,7 @@
 		    $nbrcpt++;
 		}
 	    }
-	    $/ = "\n";
+	    local $/ = "\n";
 	    close BODY;
 	}
     }
@@ -162,7 +162,7 @@
 
     ## Le champ separateur de paragraphe est un ensemble
     ## de lignes vides
-    $/ = '';
+    local $/ = '';
 
     ## Parcour du bounce, paragraphe par paragraphe
     foreach (<$fic>) {
@@ -175,7 +175,7 @@
 
 	    ## Parcour du paragraphe
 	    my @paragraphe = split /\n/, $_;
-	    $/ = "\n";
+	    local $/ = "\n";
 	    my ($champ_courant, %champ);
 	    foreach (@paragraphe) {
 
@@ -194,7 +194,7 @@
 		$$from = $1;
 		
 	    }
-	    $/ = '';
+	    local $/ = '';
 	    
 	    $champ{from} =~ s/^.*<(.+)[\>]$/$1/;
 	    $champ{from} =~  y/[A-Z]/[a-z]/;
@@ -225,7 +225,7 @@
 	    
 	    ## Parcour du paragraphe
 	    my @paragraphe = split /\n/, $_;
-	    $/ = "\n";
+	    local $/ = "\n";
 	    foreach (@paragraphe) {
 		
 		if (/^(\S[^\(]*)/) {
@@ -242,7 +242,7 @@
 		    $info{$adr}{expanded} =~ s/^[\"\<](.+)[\"\>]$/$1/;
 	        }
 	    }
-	    $/ = '';
+	    local $/ = '';
 
 	}elsif (/^\s+-+\sTranscript of session follows\s-+/m) {
 
@@ -250,7 +250,7 @@
 
 	    ## Parcour du paragraphe
 	    my @paragraphe = split /\n/, $_;
-	    $/ = "\n";
+	    local $/ = "\n";
 	    foreach (@paragraphe) {
 		
 		if (/^(\d{3}\s)?(\S+|\".*\")\.{3}\s(.+)$/) {
@@ -291,21 +291,21 @@
 		    }
 		}
 	    }
-	    $/ = '';
+	    local $/ = '';
 
         ## Rapport Compuserve	    
 	}elsif (/^Receiver not found:/m) {
 
 	    ## Parcour du paragraphe
 	    my @paragraphe = split /\n/, $_;
-	    $/ = "\n";
+	    local $/ = "\n";
 	    foreach (@paragraphe) {
 	    
 		$info{$2}{error} = $1 if /^(.*): (\S+)/;
 		$type = 3;
 
 	    }
-	    $/ = '';
+	    local $/ = '';
 
 	}elsif (/^\s*-+ Special condition follows -+/m) {
 
@@ -313,7 +313,7 @@
 	    
 	    ## Parcour du paragraphe
 	    my @paragraphe = split /\n/, $_;
-	    $/ = "\n";
+	    local $/ = "\n";
 	    foreach (@paragraphe) {
 	    
 		if (/^(Unknown QuickMail recipient\(s\)):/) {
@@ -329,7 +329,7 @@
 
 		}
 	    }
-	    $/ = '';
+	    local $/ = '';
 
 	}elsif (/^Your message adressed to .* couldn\'t be delivered/m) {
 
@@ -337,7 +337,7 @@
 	    
             ## Parcour du paragraphe
  	    my @paragraphe = split /\n/, $_;
- 	    $/ = "\n";
+ 	    local $/ = "\n";
 	    foreach (@paragraphe) {
 	    
 		if (/^Your message adressed to (.*) couldn\'t be delivered, for the following reason :/) {
@@ -353,7 +353,7 @@
 
 		}
              }
-             $/ = '';
+             local $/ = '';
 
          ## Rapport X400
          }elsif (/^Your message was not delivered to:\s+(\S+)\s+for the following reason:\s+(.+)$/m) {
@@ -406,7 +406,7 @@
 		 $type = 8;
 
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 ## Sendmail
 	 }elsif (/^Your message was not delivered to the following recipients:/m) {
@@ -496,7 +496,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 		 
 		 if (/^Rejected-For: (\S+),/) {
@@ -511,7 +511,7 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/^\s*-+Message not delivered to the following:/) {
 
@@ -523,7 +523,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 		 
 		 if (/^\s*(\S+)\s+(.*)$/) {
@@ -533,13 +533,13 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/unable to deliver following mail to recipient\(s\):/m) {
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 		 
 		 if (/^\d+ <(\S+)>\.{3} (.+)$/) {
@@ -549,7 +549,7 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 ## Rapport de Yahoo dans paragraphe suivant
 	 }elsif (/^Unable to deliver message to the following address\(es\)/m) {
@@ -579,7 +579,7 @@
 	     
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 		 
 		 if (/<(\S+)>\.{3} (.*)$/) {
@@ -589,7 +589,7 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/^Your message has encountered delivery problems\s+to local user \S+\.\s+\(Originally addressed to (\S+)\)/m or /^Your message has encountered delivery problems\s+to (\S+)\.$/m or /^Your message has encountered delivery problems\s+to the following recipient\(s\):\s+(\S+)$/m) {
 
@@ -650,7 +650,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/<(\S+)>\s+(.*)$/) {
@@ -659,7 +659,7 @@
 		     $type = 26;
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 ## Rapport de AltaVista Mail dans paragraphe suivant
 	 }elsif (/unable to deliver mail to the following recipient\(s\):/m) {
@@ -711,7 +711,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/<(\S+)>\s+(.*)$/) {
@@ -720,7 +720,7 @@
 		     $type = 30;
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/^The following recipients haven\'t received this message:/m) {
 
@@ -732,7 +732,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/(\S+)$/) {
@@ -741,7 +741,7 @@
 		     $type = 31;
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/^The following destination addresses were unknown/m) {
 
@@ -753,7 +753,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/<(\S+)>/) {
@@ -762,13 +762,13 @@
 		     $type = 32;
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/^-+Transcript of session follows\s-+$/m) {
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/^(\S+)$/) {
@@ -783,7 +783,7 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 ## Rapport Bigfoot
 	  }elsif (/^The message you tried to send to <(\S+)>/m) {
@@ -813,7 +813,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/(\S+)/) {
@@ -822,7 +822,7 @@
 		     $type = 37;
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/^This Message was undeliverable due to the following reason:/m) {
 
@@ -834,7 +834,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/\s+Recipient:\s+<(\S+)>/) {
@@ -849,7 +849,7 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
 	 }elsif (/Your message could not be delivered to:/m) {
 
@@ -902,7 +902,7 @@
 
 	     ## Parcour du paragraphe
 	     my @paragraphe = split /\n/, $_;
-	     $/ = "\n";
+	     local $/ = "\n";
 	     foreach (@paragraphe) {
 
 		 if (/\s+Recipient address:\s+(\S+)/) {
@@ -918,7 +918,7 @@
 
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 
          ## Rapport MDaemon
 	 }elsif (/^(\S+) - (no such user here)\.$/m) {
@@ -946,7 +946,7 @@
 		     $info{$addr}{error} = $error;
 		 }
 	     }
-	     $/ = '';
+	     local $/ = '';
 	 ## Wanadoo    
 	 }elsif (/^(\S+); Action: Failed; Status: \d.\d.\d \((.*)\)/m) {
 	     $info{$1}{error} = $2;
