diff --git a/sympa-6.1.9-src/wwsympa/wwsympa.fcgi.in b/sympa-6.1.9-src/wwsympa/wwsympa.fcgi.in
index 5c03aa3..0ac9cbf 100644
--- a/sympa-6.1.9-src/wwsympa/wwsympa.fcgi.in
+++ b/sympa-6.1.9-src/wwsympa/wwsympa.fcgi.in
@@ -10303,9 +10303,9 @@ sub do_rename_list {
      }
      
      # set list status to pending if creation list is moderated
-     if ($param->{'status'} eq 'pending') {
-	 &report::notice_report_web('pending_list',{},$param->{'action'},$list);
-     }
+#     if ($param->{'status'} eq 'pending') {
+#	 &report::notice_report_web('pending_list',{},$param->{'action'},$list);
+#     }
      
      if ($in{'new_robot'} eq '$robot') {
 	 $param->{'redirect_to'} = "$param->{'base_url'}$param->{'path_cgi'}/admin/$in{'new_listname'}";
