diff --git a/sympa-6.1.22-src/po/de.po b/sympa-6.1.22-src/po/de.po
index f3d84f9..911ece1 100755
--- a/sympa-6.1.22-src/po/de.po
+++ b/sympa-6.1.22-src/po/de.po
@@ -10919,6 +10919,11 @@ msgid ""
 "Visit  %3/oauth_check/%2 to check the token state and renew it if necessary."
 msgstr ""
 
+
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr "Die von Ihnen eingegebene Passwort stimmt nicht mit diesen Seiten Normen der Stärke:% 1. Bitte wählen Sie eine stärkere Passwort."
+
 #~ msgid "The configuration file contains errors.\n"
 #~ msgstr "Diese Konfigurationsdatei enthält Fehler.\n"
 
diff --git a/sympa-6.1.22-src/po/el.po b/sympa-6.1.22-src/po/el.po
index e89be0b..33b636f 100755
--- a/sympa-6.1.22-src/po/el.po
+++ b/sympa-6.1.22-src/po/el.po
@@ -11012,6 +11012,10 @@ msgstr ""
 msgid "%s / %s"
 msgstr ""
 
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr "Ο κωδικός πρόσβασης που πληκτρολογήσατε δεν ταιριάζει με αυτό sites πρότυπα της δύναμης:% 1. Παρακαλώ επιλέξτε μια ισχυρότερη κωδικό πρόσβασης."
+
 #, fuzzy
 #~ msgid "The configuration file contains errors.\n"
 #~ msgstr "Το αρχείο διαμόρφωσης --CONFIG-- περιέχει λάθη.\n"
diff --git a/sympa-6.1.22-src/po/es.po b/sympa-6.1.22-src/po/es.po
index 557f0bc..b8d32c9 100755
--- a/sympa-6.1.22-src/po/es.po
+++ b/sympa-6.1.22-src/po/es.po
@@ -11170,6 +11170,10 @@ msgstr ""
 msgid "Plugin"
 msgstr ""
 
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr "La contraseña que escribió no coincide con estos sitios niveles de fuerza:% 1. Por favor, escoja una contraseña fuerte."
+
 #, fuzzy
 #~ msgid "The configuration file contains errors.\n"
 #~ msgstr "El archivo de configuración --CONFIG-- contiene errores.\n"
diff --git a/sympa-6.1.22-src/po/fr.po b/sympa-6.1.22-src/po/fr.po
index e40d819..45c9a25 100755
--- a/sympa-6.1.22-src/po/fr.po
+++ b/sympa-6.1.22-src/po/fr.po
@@ -10474,6 +10474,10 @@ msgstr "modèle d'alias de liste"
 msgid "no mail (useful for vacations)"
 msgstr "interrompre la réception des messages (utile pendant les vacances)"
 
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr "Le mot de passe que vous avez saisi ne correspond pas à ces sites normes de résistance:% 1. S'il vous plaît choisir un mot de passe fort."
+
 #~ msgid "The configuration file contains errors.\n"
 #~ msgstr "Le fichier de configuration contient des erreurs.\n"
 
diff --git a/sympa-6.1.22-src/po/it.po b/sympa-6.1.22-src/po/it.po
index d0c355f..28a78ed 100755
--- a/sympa-6.1.22-src/po/it.po
+++ b/sympa-6.1.22-src/po/it.po
@@ -11363,6 +11363,10 @@ msgstr ""
 msgid "Searches are limited to 50 results, except when using a @domain mask."
 msgstr ""
 
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr "La password digitata non corrisponde a questi siti norme di forza:% 1. Si prega di scegliere una password forte."
+
 # nlsref 1,1
 #, fuzzy
 #~ msgid "The configuration file contains errors.\n"
diff --git a/sympa-6.1.22-src/po/pt.po b/sympa-6.1.22-src/po/pt.po
index e131b47..dd68b0e 100755
--- a/sympa-6.1.22-src/po/pt.po
+++ b/sympa-6.1.22-src/po/pt.po
@@ -11928,6 +11928,10 @@ msgid ""
 "was over the maximum size allowed on this list."
 msgstr ""
 
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr "A senha que você digitou não corresponde esse Sites padrões de força:% 1. Por favor escolha uma senha forte."
+
 #, fuzzy
 #~ msgid "URL of main Web page"
 #~ msgstr "Página de Administração %1"
diff --git a/sympa-6.1.22-src/po/sympa.pot b/sympa-6.1.22-src/po/sympa.pot
index 81e0251..cc5a732 100644
--- a/sympa-6.1.22-src/po/sympa.pot
+++ b/sympa-6.1.22-src/po/sympa.pot
@@ -8691,3 +8691,13 @@ msgstr ""
 #: ../wwsympa/wwslib.pm:39
 msgid "no mail (useful for vacations)"
 msgstr ""
+
+#: default/mail_tt2/message_report.tt2:59
+msgid "List size has exceded the limit, %1, set by the administrators."
+msgstr ""
+
+#: ../default/web_tt2/error.tt2:77
+msgid "The password you typed does not match this sites standards of strength: %1. Please pick a stronger password."
+msgstr ""
+
+
diff --git a/sympa-6.1.22-src/soap/sympasoap.pm b/sympa-6.1.22-src/soap/sympasoap.pm
index bfc6d7c..58e5309 100644
--- a/sympa-6.1.22-src/soap/sympasoap.pm
+++ b/sympa-6.1.22-src/soap/sympasoap.pm
@@ -768,7 +768,22 @@ sub add {
 	$u->{'email'} = $email;
 	$u->{'gecos'} = $gecos || $u2->{'gecos'};
 	$u->{'date'} = $u->{'update_date'} = time;
-	$u->{'password'} = $u2->{'password'} || &tools::tmp_passwd($email) ;
+
+	# If Password validation is enabled check the submitted password 
+  # against the site configured constraints
+	if ($u2->{'password'}) {
+		if (my $result = &tools::password_validation($u->{'password'})) {
+			&Log::do_log('info', 'add %s@%s %s from %s : scenario error', $listname,$robot,$email,$sender);
+			die SOAP::Fault->faultcode('Server')
+		    ->faultstring('Weak password')
+				->faultdetail('Weak password: '. $result);
+		}
+		$u->{'password'} = $u2->{'password'};
+	}
+	else {
+		$u->{'password'} = &tools::tmp_passwd($email) ;
+	}
+
 	$u->{'lang'} = $u2->{'lang'} || $list->{'admin'}{'lang'};
 	
 	unless ($list->add_user($u)) {
diff --git a/sympa-6.1.22-src/src/lib/confdef.pm b/sympa-6.1.22-src/src/lib/confdef.pm
index 7b471dc..19aadfc 100644
--- a/sympa-6.1.22-src/src/lib/confdef.pm
+++ b/sympa-6.1.22-src/src/lib/confdef.pm
@@ -1571,6 +1571,13 @@ our @params = (
         'edit'    => '1',
         'file'    => 'sympa.conf',
     },
+    {
+        'name'    => 'password_validation',
+        'query'   => 'The password validation techniques to be used against user passwords that are added to mailing lists. Options come from Data::Password (http://search.cpan.org/~razinf/Data-Password-1.07/Password.pm#VARIABLES), ie password_validation MINLEN=8,GROUPS=3,DICTIONARY=4,DICTIONARIES=/pentest/dictionaries',
+        'default' => undef,
+        'edit'    => '1',
+        'file'    => 'sympa.conf',
+    },
 
 ## Not implemented yet.
 ##    {
diff --git a/sympa-6.1.22-src/src/lib/tools.pm b/sympa-6.1.22-src/src/lib/tools.pm
index 37bd999..e0af65f 100644
--- a/sympa-6.1.22-src/src/lib/tools.pm
+++ b/sympa-6.1.22-src/src/lib/tools.pm
@@ -4048,4 +4048,28 @@ sub decode_header {
     }
 }
 
+sub password_validation {
+	my ($password) = @_;
+	my $pv = $Conf::Conf{'password_validation'};
+	if ($pv && $password) {
+		use Data::Password qw(:all);
+		use Switch;
+		my @techniques = split(/,/, $pv);
+		foreach my $technique (@techniques) {
+			my ($key, $value) = $technique =~ /([^=]+)=(.*)/;
+			switch ($key) {
+				case 'DICTIONARY' { $DICTIONARY = $value; }
+				case 'FOLLOWING' { $FOLLOWING = $value; }
+				case 'GROUPS' { $GROUPS = $value; }
+				case 'MINLEN' { $MINLEN = $value; }
+				case 'MAXLEN' { $MAXLEN = $value; }
+				# TODO: How do we handle a list of dictionaries?
+				case 'DICTIONARIES' { push @DICTIONARIES, $value; }
+			}
+		}
+		return IsBadPassword($password);
+	}
+	return undef;
+}
+
 1;
diff --git a/sympa-6.1.22-src/src/sympa_wizard.pl.in b/sympa-6.1.22-src/src/sympa_wizard.pl.in
index cb886fd..c4d4c98 100644
--- a/sympa-6.1.22-src/src/sympa_wizard.pl.in
+++ b/sympa-6.1.22-src/src/sympa_wizard.pl.in
@@ -324,6 +324,12 @@ sub check_cpan {
 						 package_name => 'Crypt-CipherSaber',
 						 usage => 'this module provides reversible encryption of user passwords in the database.  Useful when updating from old version with password reversible encryption, or if secure session cookies in non-SSL environments are required.',
 						},
+			'Data::Password' => {
+				      required_version =>'1.07',
+				      package_name => 'Data-Password',
+				      mandatory => 0,
+				      usage => 'Used for configureable hardening of passwords via the password_validation sympa.conf directive.',
+				     },
 			'DB_File' => {
 				      required_version =>'1.75',
 				      package_name => 'DB_File',
diff --git a/sympa-6.1.22-src/web_tt2/error.tt2 b/sympa-6.1.22-src/web_tt2/error.tt2
index 65d97b8..3cd16b5 100644
--- a/sympa-6.1.22-src/web_tt2/error.tt2
+++ b/sympa-6.1.22-src/web_tt2/error.tt2
@@ -74,6 +74,7 @@
   [% ELSIF u_err.msg == 'user_already_subscriber' %][%|loc(u_err.email,u_err.list)%]%1 is already subscribed to the list %2[%END%] 
   [% ELSIF u_err.msg == 'no_passwd' %][%|loc%]Please provide your password[%END%]
   [% ELSIF u_err.msg == 'diff_passwd' %][%|loc%]The passwords you typed do not match[%END%]
+  [% ELSIF u_err.msg == 'passwd_validation' %][%|loc(u_err.reason)%]The password you typed does not match this sites standards of strength: %1. Please pick a stronger password.[%END%]
   [% ELSIF u_err.msg == 'wrong_input_path' %][%|loc(u_err.tpl)%]Provided path is incorrect for template '%1'[%END%]
   [% ELSIF u_err.msg == 'cannot_open_file' %][%|loc(u_err.path)%]Cannot open file '%1'[%END%]
   [% ELSIF u_err.msg == 'listname_needed' %][%|loc%]You need to provide list name[%END%]
diff --git a/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in b/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in
index 5c5af41..8b366b0 100644
--- a/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in
+++ b/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in
@@ -4961,6 +4961,15 @@ sub do_subscribe {
 		$u->{'gecos'} = $param->{'user'}{'gecos'} || $in{'gecos'};
 		$u->{'date'} = $u->{'update_date'} = time;
 		$u->{'password'} = $param->{'user'}{'password'};
+		if (my $reason = &tools::password_validation($u->{'password'})) {
+			&report::reject_report_web('user','passwd_validation',{'reason' => $reason},$param->{'action'});
+			&wwslog('info','do_setpasswd: password validation');
+			&web_db_log({
+				'status' => 'error',
+			  'error_type' => 'bad_parameter'
+			});
+			return undef;
+		 }
 		$u->{'custom_attribute'} = $xml_custom_attribute if (defined $xml_custom_attribute);
 		$u->{'lang'} = $param->{'user'}{'lang'} || $param->{'lang'};
 		
@@ -5497,6 +5506,14 @@ sub unsubscribe {
 	 return undef;
      }
 
+	if (my $reason = &tools::password_validation($in{'newpasswd1'})) {
+		&report::reject_report_web('user','passwd_validation',{'reason' => $reason},$param->{'action'});
+		 &wwslog('info','do_setpasswd: password validation');
+		 &web_db_log({'status' => 'error',
+			  'error_type' => 'bad_parameter'});
+		 return undef;
+	 }
+
      if (&List::is_user_db($param->{'user'}{'email'})) {
 
 	 unless ( &List::update_user_db($param->{'user'}{'email'}, {'password' => $in{'newpasswd1'},'wrong_login_count' => 0} )) {
@@ -6400,6 +6417,14 @@ sub do_skinsedit {
 	 $u->{'gecos'} = $user{$email} || $u2->{'gecos'};
 	 $u->{'date'} = $u->{'update_date'} = time;
 	 $u->{'password'} = $u2->{'password'} || &tools::tmp_passwd($email) ;
+	 if (my $reason = &tools::password_validation($u->{'password'})) {
+		 &report::reject_report_web('user','passwd_validation',{'reason' => $reason},$param->{'action'});
+		 &wwslog('info','do_setpasswd: password validation');
+		 &web_db_log({'status' => 'error',
+			  'error_type' => 'bad_parameter'});
+		 return undef;
+	 }
+
 	 $u->{'lang'} = $u2->{'lang'} || $list->{'admin'}{'lang'};
 	 if ($comma_emails) {
 	     $comma_emails = $comma_emails .','. $email;
diff --git a/sympa-6.1.22-src/src/lib/confdef.pm b/sympa-6.1.22-src/src/lib/confdef.pm
index 19aadfc..7154dea 100644
--- a/sympa-6.1.22-src/src/lib/confdef.pm
+++ b/sympa-6.1.22-src/src/lib/confdef.pm
@@ -1577,6 +1577,7 @@ our @params = (
         'default' => undef,
         'edit'    => '1',
         'file'    => 'sympa.conf',
+        'optional' => '1',
     },
 
 ## Not implemented yet.
diff --git a/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in b/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in
index 4e0f580..156ebeb 100644
--- a/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in
+++ b/sympa-6.1.22-src/wwsympa/wwsympa.fcgi.in
@@ -4965,15 +4965,6 @@ sub do_subscribe {
 		$u->{'gecos'} = $param->{'user'}{'gecos'} || $in{'gecos'};
 		$u->{'date'} = $u->{'update_date'} = time;
 		$u->{'password'} = $param->{'user'}{'password'};
-		if (my $reason = &tools::password_validation($u->{'password'})) {
-			&report::reject_report_web('user','passwd_validation',{'reason' => $reason},$param->{'action'});
-			&wwslog('info','do_setpasswd: password validation');
-			&web_db_log({
-				'status' => 'error',
-			  'error_type' => 'bad_parameter'
-			});
-			return undef;
-		 }
 		$u->{'custom_attribute'} = $xml_custom_attribute if (defined $xml_custom_attribute);
 		$u->{'lang'} = $param->{'user'}{'lang'} || $param->{'lang'};
 		
@@ -5515,7 +5506,7 @@ sub unsubscribe {
 
 	if (my $reason = &tools::password_validation($in{'newpasswd1'})) {
 		&report::reject_report_web('user','passwd_validation',{'reason' => $reason},$param->{'action'});
-		 &wwslog('info','do_setpasswd: password validation');
+		 &wwslog('info','password validation error: '. $reason);
 		 &web_db_log({'status' => 'error',
 			  'error_type' => 'bad_parameter'});
 		 return undef;
@@ -6425,14 +6416,6 @@ sub do_skinsedit {
 	 $u->{'gecos'} = $user{$email} || $u2->{'gecos'};
 	 $u->{'date'} = $u->{'update_date'} = time;
 	 $u->{'password'} = $u2->{'password'} || &tools::tmp_passwd($email) ;
-	 if (my $reason = &tools::password_validation($u->{'password'})) {
-		 &report::reject_report_web('user','passwd_validation',{'reason' => $reason},$param->{'action'});
-		 &wwslog('info','do_setpasswd: password validation');
-		 &web_db_log({'status' => 'error',
-			  'error_type' => 'bad_parameter'});
-		 return undef;
-	 }
-
 	 $u->{'lang'} = $u2->{'lang'} || $list->{'admin'}{'lang'};
 	 if ($comma_emails) {
 	     $comma_emails = $comma_emails .','. $email;
