--- tmp/src/List.pm	2007-12-08 08:26:19.000000000 -0800
+++ src/List.pm	2007-12-08 08:29:04.803502838 -0800
@@ -2601,6 +2601,9 @@
 	$hdr->add($1, $2) if ($i=~/^([\S\-\:]*)\s(.*)$/);
     }
     
+    ## Add Sender: Header for DKIM compliance
+    $hdr->add('Sender', $name.&Conf::get_robot_conf($robot, 'return_path_suffix').'@'.$host);
+    
     ## Add RFC 2919 header field
     if ($hdr->get('List-Id')) {
 	&do_log('notice', 'Found List-Id: %s', $hdr->get('List-Id'));
